python/clx/features.py

```diff
@@ -0,0 +1,127 @@
+# Copyright (c) 2021, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import cudf
+import pandas as pd
+
+
+def binary(dataframe, entity_id, feature_id):
+    """
+    Create binary feature dataframe using provided dataset, entity, and feature.
+
+    :param values: dataframe
+    :type values: cudf.DataFrame
+    :param values: entity_id
+    :type values: str
+    :param values: feature_id
+    :type values: str
+    :return: dataframe
+    :rtype: cudf.DataFrame
+
+    Examples
+    --------
+    >>> import cudf
+    >>> import clx.features
+    >>> df = cudf.DataFrame(
+            {
+                "time": [1, 2, 3],
+                "user": ["u1", "u2", "u1",],
+                "computer": ["c1", "c1", "c3"],
+            }
+        )
+    >>> output = clx.features.binary(df, "user", "computer")
+    >>> output
+            c1	c3
+        user
+        u1	1.0	1.0
+        u2	1.0	0.0
+    """
+    if entity_id and feature_id not in dataframe.columns:
+        raise Exception(
+            "{0} and {1} must be column names in the input dataframe".format(
+                entity_id, feature_id
+            )
+        )
+    df_grouped = (
+        dataframe.groupby([entity_id, feature_id])
+        .count()
+        .reset_index()
+        .set_index(entity_id)
+    )
+    # https://github.com/rapidsai/cudf/issues/1214
+    pdf_grouped = df_grouped.to_pandas()
```

Comment:
It looks like you're doing the aggregations separately? If that's the case, could you use `pivot` to do this? https://github.com/rapidsai/cudf/pull/5779

Reply:
Sure I can give `pivot` a try. Thanks!